src/modules/mc_att_control/mc_att_control_main.cpp

```diff
@@ -994,9 +993,28 @@ MulticopterAttitudeControl::control_attitude_rates(float dt)
 	_board_rotation = board_rotation_offset * _board_rotation;
 
 	// get the raw gyro data and correct for thermal errors
-	math::Vector<3> rates(_sensor_gyro.x * _sensor_correction.gyro_scale[0] + _sensor_correction.gyro_offset[0],
-			      _sensor_gyro.y * _sensor_correction.gyro_scale[1] + _sensor_correction.gyro_offset[1],
-			      _sensor_gyro.z * _sensor_correction.gyro_scale[2] + _sensor_correction.gyro_offset[2]);
+	math::Vector<3> rates;
+
+	if (_sensor_correction.gyro_mapping[_selected_gyro] == 0) {
```

Comment:
That's correct - there is no guarantee at all that this matches between calibration time and runtime.